### ISSUE TITLE
Compatibility with column-enforce-mode and long literate (latex-style) comments

### ISSUE THREAD

**Issue by listx** [haskell/haskell-mode#331]
`column-enforce-mode` (https://github.com/jordonbiondo/column-enforce-mode) is a minor mode that highlights long lines (starting with the 80th column).

As of today, `column-enforce-mode` has a new feature variable called `column-enforce-comments` that disables highlighting long lines if they are comment lines. From what I can tell, haskell-mode has a caching mechanism for applying `font-lock-comment-face` (in `haskell-fl-latex-comments`) and `column-enforce-comments` does not work reliably with it (sometimes it highlights long comment lines, sometimes it doesn't --- all within the same file).

Is there a way to fix this?

**Comment by gracjan** [comment]
@listx: I know this is an old issue, but could you create a test case in `tests/haskell-font-lock-tests.el`? Thanks.

**Comment by listx** [comment]
Between work and my other projects I do not have the time to learn how the test suite works, and to add a test case for it. Sorry!

FWIW the issue is still relevant.

**Comment by gracjan** [comment]
Note that we have changed significantly how comments are marked in Latex sources. It used to use start and and of comment, right now it is per line. Can you check if the issue really persist after #1050 got merged?

**Comment by listx** [comment]
So, my last comment stating that the issue still persists was in error. Since the time of the original report, I had moved to MMM (Multiple Major Modes) and had enabled LaTeX mode for the latex-style comment regions. For some reason I had equated the long-line highlighting within the LaTeX mode (not `literate-haskell-mode`) parts with this bug report.

Simply enabling `literate-haskell-mode` only reveals that indeed this issue has been fixed (probably after #1050 got merged as you suggested). The long lines in the latex-style comment blocks do not get highlighted by column-enforce-mode for me.

Thanks @gracjan for the tip
